### PR TITLE
Rework pylon loadout attribute

### PR DESCRIPTION
**Pull request by neilzar**
**When merged this pull request will:**
- Improve performance of the Loadout attribute, 75% improvement
- Always give the option for gunner/pilot when there is a gunner seat, not just when there is a unit in the seat.

### CODE REVIEW

**Comment by neilzar** [comment]
Works fine when testing in Multiplayer alone, unsure whether removing the locality check for `setPylonLoadout` breaks it; I have never had to remote execute it but that could have been circumstantial.

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
+		Kex, CoxLedere, CreepPork_LV, NeilZar
+
+	DESCRIPTION:
+		Opens the "ammo" dialog for vehicles.
+
+	Parameters:
+	   _this:  OBJECT - Vehicle to edit
+
+	Returns:
+	   none
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: Is it then none or nothing?

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf [hunk 2]
```diff
@@ -53,25 +56,16 @@ if (_hasGunner) then
 {
 	_addWeaponsTo = [[], [0]] select (_dialogResult select 0 == 1);
 	_plane setVariable ["Achilles_var_changePylonAmmo_Assigned", _addWeaponsTo, true];
+	_dialogResult deleteAt 0
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: ;

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf (hunk 1 above)
Comment (neilzar) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: Kex has been using none so far.

Comment (CreepPork) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: https://github.com/ArmaAchilles/Achilles/pull/437/files/350697bf048cd82728c0d66bd1573fb5955632f7#r249569240

File: @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf [hunk 3]
```diff
@@ -1,47 +1,50 @@
...
 private _planeType = typeOf _plane;
-if (!isClass (configFile >> "cfgVehicles" >> _planeType >> "Components" >> "TransportPylonsComponent")) exitWith {[localize "STR_AMAE_NO_DYNAMIC_LOADOUT"] call Achilles_fnc_ShowZeusErrorMessage; nil};
-
-private _hasGunner = not isNull gunner _plane;
+if (!isClass (configFile >> "cfgVehicles" >> _planeType >> "Components" >> "TransportPylonsComponent")) exitWith { [localize "STR_AMAE_NO_DYNAMIC_LOADOUT"] call Achilles_fnc_ShowZeusErrorMessage; };
 
+private _hasGunner = !(fullCrew [_plane, "gunner", true] isEqualTo []);
 private _allCurrentPylonMagazines = getPylonMagazines _plane;
-private _pylon_cfgs = (configFile >> "cfgVehicles" >> _planeType >> "Components" >> "TransportPylonsComponent" >> "pylons") call BIS_fnc_returnChildren;
+private _pylonCfgs = configProperties [configFile >> "cfgVehicles" >> _planeType >> "Components" >> "TransportPylonsComponent" >> "pylons", "isClass _x"];
```
Comment (oOKexOo) [inline @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: `BIS_fnc_returnChildren` is indeed rather inefficient, that's why we introduced `Achilles_fnc_returnChildren`. Your approach could be even faster. How about using `Achilles_fnc_returnChildren` and integrating your approach by changing `Achilles_fnc_returnChildren`?

Comment (neilzar) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: I will do that after my midterms. I know from testing that this approach is about 5-6x faster than `Achilles_fnc_returnChildren`.

Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: I would still use `Achilles_fnc_returnChildren` and improve the function with your approach, such that other parts of Achilles can also profit from it.

Comment (neilzar) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: That was already changed...

Comment (oOKexOo) [reply @AresModAchillesExpansion/addons/functions_f_achilles/functions/changeAttributes/fn_changePylonAmmo.sqf]: My bad. Not used to mobile view.

**Comment by CreepPork** [comment]
👍 Looks good to merge @neilzar.

**Pull request merged**